### PR TITLE
Return list of addresses for contact and companies

### PR DESCRIPTION
**Pull request by jenssegers**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -465,7 +465,15 @@ Get a list of contacts.
...
                 + avatar (string)
-                + address (Address)
+                + addresses (array)
+                    + (object)
+                        + type: `primary` (enum[string])
+                            + Members
+                                + primary
+                                + invoice
+                                + delivery
+                                + visitor
```
Comment (stivni) [inline apiary.apib]: PP : `visitor` seems like a weird label to me...

Comment (stivni) [reply apiary.apib]: same with `invoice` btw... I think we should call it `invoicing` address...

Comment (stivni) [reply apiary.apib]: also, thinking about it further: does have a `delivery` or `shipping` address makes sense? If a company has 6 addresses, why wouldn't all of them be able to be a `shipping` or `visiting` address?

Comment (jenssegers) [reply apiary.apib]: I tried to not mix `-ing` label. `delivering`, `visiting`, `invoicing` also sounds a bit weird.

Comment (mrnnn) [reply apiary.apib]: that's a big PP, i think they sound better :-)

But yeah, the address types. It'd be nice if we could just add a ton of named address and set one as "default invoicing address" for example.  That would fix everything.

Comment (jenssegers) [reply apiary.apib]: In the end we only need `primary` and `invoicing` right?

Comment (stivni) [reply apiary.apib]: > In the end we only need primary and invoicing right? The others are just other with a label.

or addresses could even have multiple purposes...
the most difficult part is that today a lot of our code will assume that addresses of certain types might exist **and** assume there is only one...

so I don't think we are able to shit to `other` straight away...

File: apiary.apib [hunk 2]
```diff
@@ -603,10 +607,14 @@ Add a new contact.
...
-        + delivery_address (Addressee, optional)
-        + visiting_address (Addressee, optional)
-        + invoice_address (Addressee, optional)
+        + addresses (array, optional)
+            + (object)
+                + type: `primary` (enum[string], required)
+                    + Members
+                        + primary
+                        + invoicing
+                        + other
```
Comment (jenssegers) [inline apiary.apib]: Add back `delivery` and `visiting`

**Pull request merged**